### PR TITLE
[All Products] implement 

### PR DESCRIPTION
**Pull request by PhuongTT**

### CODE REVIEW

File: src/app/(MainLayout)/boards/components/AllProducts/components/ItemCatalog.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+import { Icon } from '@/components';
+type ProductItem = {
+  data: Product;
+  viewDetail?: boolean;
+};
+
+export function ItemCatalog({ data, viewDetail = true }: ProductItem) {
+  return (
+    <>
```
Comment (tuanna7494) [inline src/app/(MainLayout)/boards/components/AllProducts/components/ItemCatalog.tsx:16]: Remove and rename AllProducts => ProductItem => and move all file in components to ProductItem

File: src/app/(MainLayout)/boards/components/ProductsDetail/components/ItemDetailVideo.tsx [hunk 2]
```diff
@@ -0,0 +1,16 @@
+'use client';
+
+import { Product } from '@/types';
+import { Flex } from 'antd';
+import ReactPlayer from 'react-player';
```
Comment (tuanna7494) [inline src/app/(MainLayout)/boards/components/ProductsDetail/components/ItemDetailVideo.tsx:5]: break line

File: src/app/(MainLayout)/boards/components/ProductsDetail/components/ItemDetailImage.tsx [hunk 3]
```diff
@@ -0,0 +1,23 @@
+'use client';
```
[the remark this answers is not in the document]
Comment (tuanna7494) [reply src/app/(MainLayout)/boards/components/ProductsDetail/components/ItemDetailImage.tsx:1]: Remove folder components
Shouldn't make this

File: src/app/(MainLayout)/boards/components/ProductsDetailModal/index.tsx [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      footer={[]}
+    >
+      <Flex vertical gap={8} className='h-full flex-1 overflow-hidden'>
+        <Flex className='mt-6 gap-6 px-6 [&_.ant-typography]:text-white'>
+          {ACTION_MENU.map((item: MenuItemProps) => (
+            <MenuItem key={`${item.label}-${item.id}`} {...item} />
+          ))}
+        </Flex>
+        <Flex vertical justify='center' align='center' className='flex-1 overflow-auto p-6'>
+          {data.type === ProductType.TEXT && <ItemDetailText data={data} />}
```
Comment (tuanna7494) [inline src/app/(MainLayout)/boards/components/ProductsDetailModal/index.tsx:64]: Should be make only a ProductDetailItem file

File: src/app/(MainLayout)/boards/components/AllProducts/components/ItemCatalog.tsx [hunk 5]
```diff
@@ -0,0 +1,43 @@
+'use client';
+
+import { Product } from '@/types';
+import { Checkbox, Flex } from 'antd';
+import clsx from 'clsx';
+import Image from 'next/image';
+
+import { Icon } from '@/components';
```
Comment (tuanna7494) [inline src/app/(MainLayout)/boards/components/AllProducts/components/ItemCatalog.tsx:8]: break line

File: src/app/(MainLayout)/boards/components/AllProducts/components/ItemCatalog.tsx [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+type ProductItem = {
```
Comment (tuanna7494) [inline src/app/(MainLayout)/boards/components/AllProducts/components/ItemCatalog.tsx:9]: Should be use inteface for obj

File: src/app/(MainLayout)/boards/components/AllProducts/components/ItemCatalog.tsx (hunk 1 above)
Comment (tuanna7494) [reply src/app/(MainLayout)/boards/components/AllProducts/components/ItemCatalog.tsx:16]: Rename ItemCatalog => ProductItemCatelog

File: src/app/(MainLayout)/boards/components/ProductsDetail/components/ItemDetailVideo.tsx (hunk 2 above)
Comment (PhuongTT) [reply src/app/(MainLayout)/boards/components/ProductsDetail/components/ItemDetailVideo.tsx]: fixed

File: src/app/(MainLayout)/boards/components/AllProducts/index.tsx [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+
+type ProductItem = {
+  products: Product[];
+};
+
+export function AllProducts({ products }: ProductItem) {
+  return (
+    <Masonry columnsCount={2} gutter='24px'>
+      {products.map((item, i) => {
+        if (item.type === ProductType.TEXT) {
```
[the remark this answers is not in the document]
Comment (PhuongTT) [reply src/app/(MainLayout)/boards/components/AllProducts/index.tsx]: fixed

File: src/app/(MainLayout)/boards/components/ProductsDetailModal/index.tsx (hunk 4 above)
Comment (PhuongTT) [reply src/app/(MainLayout)/boards/components/ProductsDetailModal/index.tsx]: fixed

File: src/app/(MainLayout)/boards/components/AllProducts/components/ItemCatalog.tsx (hunk 5 above)
Comment (PhuongTT) [reply src/app/(MainLayout)/boards/components/AllProducts/components/ItemCatalog.tsx]: fixed

**Pull request merged**